packages/edit-site/src/components/pattern-actions/index.js

```diff
@@ -0,0 +1,297 @@
+				{ type: 'snackbar', id: 'edit-site-patterns-success' }
+			);
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: __( 'An error occurred while deleting the pattern.' );
+			createErrorNotice( errorMessage, {
+				type: 'snackbar',
+				id: 'edit-site-patterns-error',
+			} );
+		}
+	};
+
+	const deleteItem = async () => {
+		if ( isTemplateRemovable( record ) ) {
+			removeTemplate( record );
+		} else if ( isUserPattern ) {
+			deletePattern( record );
+		}
+	};
+
+	async function revertAndSaveTemplate() {
+		try {
+			await revertTemplate( record, { allowUndo: false } );
+			await saveEditedEntityRecord( 'postType', record.type, record.id );
+
+			createSuccessNotice(
+				sprintf(
+					// translators: %s: the pattern, template/part's title.
+					__( '"%s" reverted.' ),
+					decodedTitle
+				),
+				{
+					type: 'snackbar',
+					id: 'edit-site-template-reverted',
+				}
+			);
+		} catch ( error ) {
+			const fallbackErrorMessage = sprintf(
+				// translators: %s: a post type label, e.g., Template, Template Part or Pattern.
+				__( 'An error occurred while reverting the %s.' ),
+				POST_TYPE_LABELS[ record?.type ] ??
+					POST_TYPE_LABELS[ TEMPLATE_POST_TYPE ]
+			);
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: fallbackErrorMessage;
+
+			createErrorNotice( errorMessage, { type: 'snackbar' } );
+		}
+	}
+
+	const { categoryId } = getQueryArgs( window.location.href );
+
+	return (
+		<DropdownMenu
+			icon={ icon }
+			label={ __( 'Actions' ) }
+			className={ className }
+			popoverProps={ { placement: 'bottom-end', ...popoverProps } }
+			toggleProps={ {
+				describedBy: sprintf(
+					// translators: %s: the pattern, template/part's title.
+					__( 'Action menu for "%s"' ),
+					decodedTitle
+				),
+				...toggleProps,
+			} }
+		>
+			{ ( { onClose } ) => (
+				<MenuGroup>
+					{ isEditable && (
+						<RenameMenuItem
+							postId={ record?.id }
+							postType={ record?.type }
+							onClose={ onClose }
+						/>
+					) }
+					{ isDuplicable && (
+						<DuplicateMenuItem
+							categoryId={ categoryId }
+							item={ record || item }
+							onClose={ onClose }
+							label={ __( 'Duplicate' ) }
+						/>
+					) }
+					{ isUserPattern && (
+						<MenuItem onClick={ () => exportAsJSON( record ) }>
+							{ __( 'Export as JSON' ) }
+						</MenuItem>
+					) }
+					{ isEditable && (
+						<DeleteMenuItem
+							postType={ record?.type }
+							onRemove={ () => {
+								deleteItem();
+								onRemove?.( record );
+								onClose();
+							} }
+							title={ decodedTitle }
+						/>
+					) }
+					{ isRevertable && (
+						<RevertMenuItem
+							onRemove={ () => {
+								revertAndSaveTemplate();
+								onClose();
+							} }
+						/>
+					) }
+				</MenuGroup>
+			) }
+		</DropdownMenu>
+	);
+}
+
+function DeleteMenuItem( { onRemove, title } ) {
```

Comment:
Is it worth also splitting out the `DeleteMenuItem` and `RevertMenuItem` components to their own files to match `DuplicateMenuItem` & `RenameMenuItem`?